src/commands.ts

```diff
@@ -250,40 +165,21 @@ export class Commands {
 	 * Log out from the currently logged-in deployment.
 	 */
 	public async logout(): Promise<void> {
-		const url = this.mementoManager.getUrl();
-		if (!url) {
-			// Sanity check; command should not be available if no url.
-			throw new Error("You are not logged in");
-		}
-		await this.forceLogout();
-	}
-
-	public async forceLogout(): Promise<void> {
-		if (!this.contextManager.get("coder.authenticated")) {
+		if (!this.deploymentManager.isAuthenticated()) {
 			return;
 		}
+
 		this.logger.info("Logging out");
-		// Clear from the REST client.  An empty url will indicate to other parts of
-		// the code that we are logged out.
-		this.restClient.setHost("");
-		this.restClient.setSessionToken("");
 
-		// Clear from memory.
-		await this.mementoManager.setUrl(undefined);
-		await this.secretsManager.setSessionToken(undefined);
+		await this.deploymentManager.logout();
```

Comment:
I ran into some confusion between how `deploymentManager` has auth-related stuff like logout but then there is also a `loginCoordinator` and that does some auth-related stuff and it feels like they are cross-contaminated.

I think to me the deployment and its auth are one singular object, and conceptually having them split feels like too much, and it could be unclear on what the responsibilities are for each.

But, this is just me musing out loud, possibly I just need more time with it.